### ISSUE TITLE
CORS error in Browser; Different error in Firebase Console

### ISSUE THREAD

**Issue by ChaddPortwine** [googleapis/google-api-nodejs-client#1206]
**Goal:**
Use Firebase Functions to Read/Write to a sheet using Domain Wide Delegation on behalf of a G Suite admin.

**Issue:**
Getting the wrong error message in Browser.
`ERROR: No "Access-Control-Allow-Origin" header is present`
Browser will indicate a CORS issue while the Firebase console will show the actual/different error.

**Code:**
This works with no errors
```
        return new Promise((resolve, reject) => {
            sheets.spreadsheets.values.get({
                auth: jwtClient,
                spreadsheetId: 'xyz...123',
                range: 'hello!A1:B2',
              }, (err, res) => {
                if (err) {
                    reject(err);
                    return console.log('The API returned an error: ' + err);
                }
                else {
                    return res
                }
            });
        })

```
Change `.get` to `.update` to trigger an error
```
        return new Promise((resolve, reject) => {
            sheets.spreadsheets.values.update({       // <---- Changed to .update
                auth: jwtClient,
                spreadsheetId: 'xyz...123',
                range: 'hello!A1:B2',
              }, (err, res) => {
                if (err) {
                    reject(err);
                    return console.log('The API returned an error: ' + err);
                }
                else {
                    return res
                }
            });
        })

```
**Expected Errors:**
The broswer should show an error that `'valueInputOption' is required but not specified`.

**Actual Errors:**
_Error in Browser Reads_
```
Failed to load https://us-central1-development-is.cloudfunctions.net/app/getSheet: No 'Access-Control-Allow-Origin' header is present on the requested resource. 
```

_Error in Firebase Console Reads_

```
Error: 'valueInputOption' is required but not specified     at Request._callback (/user_code/node_modules/googleapis/node_modules/google-auth-library/lib/transporters.js:85:15)     at Request.self.callback (/user_code/node_modules/googleapis/node_modules/request/request.js:185:22)     at emitTwo (events.js:106:13)     at Request.emit (events.js:191:7)     at Request.<anonymous> (/user_code/node_modules/googleapis/node_modules/request/request.js:1157:10)     at emitOne (events.js:96:13)     at Request.emit (events.js:188:7)     at IncomingMessage.<anonymous> (/user_code/node_modules/googleapis/node_modules/request/request.js:1079:12)     at IncomingMessage.g (events.js:292:16)     at emitNone (events.js:91:20) | more_vert
```

**Using:**
express
firebase-functions
firebase-admin
googleapis

**Comment by JustinBeckwith** [comment]
So just to make sure I understand this all:
- The user goes to a website in the browser
- You use the Firebase functions API to do call a function
- Firebase functions uses the googleapis npm module
- You get an error from the googleapis npm module that you can console.log on the server
- The error doesn't get returned to the Firebase Admin API or the browser

Does that all ring true?  If that's the case, can you share your entire Firebase Function?  And the Code you're calling it from?

**Comment by ChaddPortwine** [comment]
Very Close

1. The user goes to a website in the browser
2. You use the Firebase functions API to do call a function
3. Firebase functions uses the googleapis npm module

If I misconfigured **one** of `sheets.spreadsheets.values...` in the Cloud Function

            sheets.spreadsheets.values.update({  //<-- change to udpate from get
                auth: noAuth,                    //<-- invaid auth or
                spreadsheetId: 'invaidId',       //<-- invaid spreadsheetId or
                range: 'outOfRange',             //<-- invaid range
              }, (err, res) => {

The Firebase functions logs (i.e., debug, info, error) return _different_ and meaningful error logs.

However, in Chrome\Inspect\Console I only ever see this error:

```
"...No 'Access-Control-Allow-Origin' header..."

```

Finally, This cloud function _can_ currently get and update. I needed to ignore the browser console error and find the corasponding Firebase Cloud Functions Logs error for trouble-shooting.

When my Firebase errors are fixed, I do not get Chrome\Inspect\Console errors. (and visa/versa)

I hope this helps. Ping me again, and I can send you the cloud function.
